src/main/java/crud/dao/impl/CarDaoJdBcImpl.java

```diff
@@ -0,0 +1,169 @@
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet set = statement.executeQuery();
+            while (set.next()) {
+                Long carId = set.getObject(1, Long.class);
+                cars.add(get(carId).get());
+            }
+            return cars;
+        } catch (SQLException throwables) {
+            throw new DataBaseException("Could not get a list of cars by driver id "
+                    + driverId, throwables);
+        }
+    }
+
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars (manufacturer_id, car_model) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = setUpdate(connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS), car)) {
+            statement.executeUpdate();
+            ResultSet set = statement.getGeneratedKeys();
+            if (set.next()) {
+                car.setId(set.getObject(1, Long.class));
+            }
+            return car;
+        } catch (SQLException throwables) {
+            throw new DataBaseException("Couldn't insert " + car + " into carsDB.", throwables);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT c.car_id, m.manufacturer_id, c.car_model, "
+                    + "m.manufacturer_name, m.manufacturer_country FROM cars c "
+                + "JOIN manufacturers m ON m.manufacturer_id = c.manufacturer_id "
+                + "WHERE c.car_id = ? AND c.deleted = FALSE";
+        String getDriversQuery = "SELECT d.driver_id, d.driver_name, d.driver_license_number "
+                    + "FROM cars c "
+                + "JOIN cars_drivers cd on c.car_id = cd.cars_id "
+                + "JOIN drivers d on d.driver_id = cd.driver_id "
+                + "WHERE c.car_id = ? AND c.deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet set = statement.executeQuery();
+            Car car = null;
+            if (set.next()) {
+                car = setCar(set);
+                PreparedStatement statement2 = connection.prepareStatement(getDriversQuery);
+                statement2.setLong(1, id);
+                ResultSet set2 = statement2.executeQuery();
+                while (set2.next()) {
+                    car.getDrivers().add(setDriver(set2));
+                }
+                statement2.close();
+            }
+            return Optional.ofNullable(car);
+        } catch (SQLException throwables) {
+            throw new DataBaseException("Could not get car by id " + id, throwables);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
```

Comment:
Same remarks as in getAllByDriver

Reply:
let's joing the mamufactures here and you will have all information about car. Later you can user getDrivers() method for each of them